lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
```

Comment:
I wrote this with the lamda first but ended up rewriting it this way because an `||= {}` seemed easier to read. Also, we are using both in the subscription key method. It's a two tier hash so we don't need to compute anything.